app/models/user.rb

```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  validates :username, presence: true, uniqueness: { case_sensitive: false }
+  validates :email, presence: true, uniqueness: { case_sensitive: false },
+            format: { with: /\A(.+)@(.+)\z/ }, length: { minimum: 4, maximum: 254 }  
+
+  validates :password, presence: true
```

Comment:
- [ ] [OPTIONAL] You did a great job validating the email, how about validating the password so it's a strong and safe one? Bonus points if you use a custom validate method and add a custom error to it.